python/sglang/srt/mem_cache/sparsity/algorithms/deepseek_nsa.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from typing import TYPE_CHECKING, Any, Optional
+
+import nvtx
+import torch
+
+from sglang.srt.mem_cache.sparsity.algorithms.base_algorithm import (
+    BaseSparseAlgorithm,
+    SparseMode,
+)
+
+if TYPE_CHECKING:
+    from sglang.srt.layers.attention.nsa.nsa_indexer import Indexer
+    from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+logger = logging.getLogger(__name__)
+
+
+class DeepSeekNSAAlgorithm(BaseSparseAlgorithm):
```

Comment:
can't we inherit it from a common TokenSparseAlgorithm class? Or if there is no much difference between token and page based algorithms, we can just have more public functions implemented in base algorithm